viewer/static/js/app/components/views/chart.js

```diff
@@ -7,125 +7,106 @@ import ChartHoverContainer from '../containers/chart-hover-container';
 import ChartFocus from './chart-focus';
 
 
-export default class extends React.Component {
-  /**
-   * sortedPopulationsToShow often updates before populationData does. That is,
-   * sortedPopulationsToShow often names populations that do not yet have
-   * corresponding data in populationData.
-   *
-   * This component can't exactly do much when that happens. It can't exactly
-   * render data that it doesn't have yet. So it's better off not re-rendering
-   * at all. As soon as populationData is updated to have data for all of the
-   * populations named in sortedPopulationsToShow, this test will pass and the
-   * component will re-render.
-   */
-  shouldComponentUpdate(nextProps) {
-    let shouldUpdate = true;
+export default function(props) {
+  if (props.noData) {
+    return (
+      <div className={`chart chart-${props.metricId} no-data`}>
+        <span className="warning">No data</span>
+        <span>(try selecting different cohorts)</span>
+      </div>
+    );
+  } else if (props.isFetching) {
+    return (
+      <div className={`chart is-fetching chart-${props.metricId}`}>
+        <Fetching />
+      </div>
+    );
+  } else {
+    // sortedPopulationsToShow, but reversed, and with populations removed that
+    // do not have corresponding data in populationData.
+    //
+    // That last part (removing populations that do not also currently have
+    // corresponding data in populationData) is important because we don't want
+    // to attempt to render a population that we don't have data for.
+    const reverseSortedAvailablePopulations = props.sortedPopulationsToShow.slice(0).reverse().filter(p => {
```

Comment:
The `slice(0)` essentially makes a copy of the array which we can reverse. That way, `props.sortedPopulationsToShow` stays in its original order. We then render the populations in reverse order so that the first population (usually control) paints over the others.

But I agree that this is expensive. Thanks for highlighting that. I'll look at doing this all in-place wiwth `reduceRight` when the populations are rendered.